src/net/tcp/stream.rs

```diff
@@ -148,6 +97,78 @@ impl TcpStream {
         }))
     }
 
+    /// Creates a new TCP stream connected to the specified address.
+    ///
+    /// This method will create a new TCP socket and attempt to connect it to the `addr`
+    /// provided. The returned future will be resolved once the stream has successfully
+    /// connected, or it will return an error if one occurs.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # fn main() -> std::io::Result<()> { async_std::task::block_on(async {
+    /// #
+    /// use async_std::net::TcpStream;
+    ///
+    /// let addr = "127.0.0.1".parse().unwrap();
+    /// let stream = TcpStream::connect_to(addr).await?;
+    /// #
+    /// # Ok(()) }) }
+    /// ```
+    pub async fn connect_to(addr: SocketAddr) -> io::Result<TcpStream> {
+        let stream = mio::net::TcpStream::connect(&addr).map(|mio_stream| {
+            #[cfg(unix)]
+            let stream = TcpStream {
+                raw_fd: mio_stream.as_raw_fd(),
+                io_handle: IoHandle::new(mio_stream),
+            };
+
+            #[cfg(windows)]
+            let stream = TcpStream {
+                // raw_socket: mio_stream.as_raw_socket(),
+                io_handle: IoHandle::new(mio_stream),
+            };
+
+            stream
+        });
+
+        enum State {
+            Waiting(TcpStream),
+            Error(io::Error),
+            Done,
+        }
+        let mut state = match stream {
+            // TODO replace with .map_or_else(State::Error, State::Waiting)
```

Comment:
Can we delete this TODO? I actually prefer the `match` statement since it's easier to see what it does at a glance, despite it being a bit more verbose than `map_or_else`. :) It takes me a moment of pause to figure out how `State::Error` is constructed in this case and how it differs from `State::Waiting`.

Reply:
Ok

Reply:
Fixed